File: assignments/crackers/personalincometax.py
options = ['Single filer', 'Married filing jointly' ,'Married filing separately', 'Head of household']
for index, value in enumerate(options):
	print(f"{index}: {value}")


choice = int(input("\nEnter filing status: "))
taxable_income = int(input("Enter taxable income: "))

tax = 0


match choice:
	case 0:	
		tax_1 = 0.10 * 8350
		second_tax = taxable_income - 8350
		if taxable_income >= 8351 and taxable_income <= 33950:
			second_tax = tax_1 + (0.15 * second_tax)
		elif taxable_income >= 33951 and taxable_income <= 82250:
			second_tax = tax_1 + (0.25 * second_tax)
		elif taxable_income >= 82251 and taxable_income <= 171550:
			second_tax = tax_1 + (0.28 * second_tax)
		elif taxable_income >= 171551 and taxable_income <= 372950:
			second_tax = tax_1 + (0.33 * second_tax)
		elif taxable_income >= 372951:
			second_tax = tax_1 + (0.35 * second_tax)
	case 1:
		tax_1 = 0.10 * 16700
		second_tax = taxable_income - 8350
		if taxable_income >= 16700 and taxable_income <= 67900:
			second_tax = tax_1 + (0.15 * second_tax)
		elif taxable_income >= 67901 and taxable_income <= 137050:
			second_tax = tax_1 + (0.25 * second_tax)
		elif taxable_income >= 137051 and taxable_income <= 208850:
			second_tax = tax_1 + (0.28 * second_tax)
		elif taxable_income >= 208851 and taxable_income <= 372950:
			second_tax = tax_1 + (0.33 * second_tax)
		elif taxable_income >= 372951:
			second_tax = tax_1 + (0.35 * second_tax)
	case 2:
		tax_1 = 0.10 * 8350
		second_tax = taxable_income - 8350
		if taxable_income >= 8351 and taxable_income <= 33950:
			second_tax = tax_1 + (0.15 * second_tax)
		elif taxable_income >= 33951 and taxable_income <= 68525:
			second_tax = tax_1 + (0.25 * second_tax)
		elif taxable_income >= 68526 and taxable_income <= 104425:
			second_tax = tax_1 + (0.28 * second_tax)
		elif taxable_income >= 104426 and taxable_income <= 186475:
			second_tax = tax_1 + (0.33 * second_tax)
		elif taxable_income >= 186476:
			second_tax = tax_1 + (0.35 * second_tax)
	case 3:
		tax_1 = 0.10 * 11950
		second_tax = taxable_income - 8350
		if taxable_income >= 11951 and taxable_income <= 45500:
			second_tax = tax_1 + (0.15 * second_tax)
		elif taxable_income >= 45501 and taxable_income <= 117450:
			second_tax = tax_1 + (0.25 * second_tax)
		elif taxable_income >= 117451 and taxable_income <= 190200:
			second_tax = tax_1 + (0.28 * second_tax)
		elif taxable_income >= 190201 and taxable_income <= 372950:
			second_tax = tax_1 + (0.33 * second_tax)
		elif taxable_income >= 372951:
			second_tax = tax_1 + (0.35 * second_tax)
	


	
print(f"Your cummulative tax is {second_tax}")












	

File: Assignment JAva/crackers/PersonalIncomeTax.java
import java.util.Scanner;

public class PersonalIncomeTax{
	public static void main(String...args){
	Scanner input = new Scanner(System.in);

		String choice = """	0: Single filer;
					1: Married filing jointly;
					2: Married filing separately;
					3: Head of household;
								""";


		System.out.print("Enter filing stasus:");
		int option = input.nextInt();

		System.out.print("Enter taxable income:");
		int taxableIncome = input.nextInt();


		int tax = 0;
		int taxRate10 = 10 / 100;
		int taxRate15 = 15 / 100;
		int taxRate25 = 25 / 100;
		int taxRate28 = 28 / 100;
		int taxRate33 = 33 / 100;
		int taxRate35 = 35 / 100;


		
		switch(choice){
			case "0":
				if(taxableIncome >= 0 && taxableIncome <= 8350)tax = taxableIncome * taxRate10;
				else if(taxableIncome >= 8351 && taxableIncome <= 33950)tax = taxableIncome * taxRate15;
				else if(taxableIncome >= 33951 && taxableIncome <= 82250)tax = taxableIncome * taxRate25;
				else if(taxableIncome >= 82251 && taxableIncome <= 171550)tax = taxableIncome * taxRate28;
				else if(taxableIncome >= 171551 && taxableIncome <= 372950)tax = taxableIncome * taxRate33;
				else if(taxableIncome >= 372951)tax = taxableIncome * taxRate35;

	
			case "1":
				if(taxableIncome >= 0 && taxableIncome <= 16700)tax = taxableIncome * taxRate10;
				else if(taxableIncome >= 16701 && taxableIncome <= 67900)tax = taxableIncome * taxRate15;
				else if(taxableIncome >= 67901 && taxableIncome <= 137050)tax = taxableIncome * taxRate25;
				else if(taxableIncome >= 137051 && taxableIncome <= 208850)tax = taxableIncome * taxRate28;
				else if(taxableIncome >= 208851 && taxableIncome <= 372950)tax = taxableIncome * taxRate33;
				else if(taxableIncome >= 372951)tax = taxableIncome * taxRate35;


			case "2":
				if(taxableIncome >= 0 && taxable_income <= 8350)tax = taxableIncome * taxRate10;
				else if(taxableIncome >= 8351 && taxable_income <= 33950)tax = taxableIncome * taxRate15;
				else if(taxableIncome >= 33951 && taxable_income <= 68525)tax = taxableIncome * taxRate25;
				else if(taxableIncome >= 68526 && taxableIncome <= 104425)tax = taxableIncome * taxRate28;
				else if(taxableIncome >= 104426 && taxableIncome <= 186375)tax = taxableIncome * taxRate33;
				else if(taxableIncome >= 186476)&& = taxableIncome * taxRate35;

			case "3":
				if(taxableIncome >= 0 && taxableIncome <= 11950)tax = taxableIncome * taxRate10;
				else if(taxableIncome >= 11951 && taxableIncome <= 45500)tax = taxableIncome * taxRate15;
				else if(taxableIncome >= 45501 && taxableIncome <= 117450)tax = taxableIncome * taxRate25;
				else if(taxableIncome >= 117451 && taxableIncome <= 190200)tax = taxableIncome * taxRate28;
				else if(taxableIncome >= 190201 && taxableIncome <= 372950)tax = taxableIncome * taxRate33;
				else if(taxableIncome >= 372951)tax = taxable_income * taxRate35;


			
		}

			System.out.printf("The tax for %d", tax);
		
	}
}